### PR TITLE
LPS-141042 export import blueprint

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-141042 - Export Blueprint

I hope this works! Petteri had mentioned that opening a new window with "blank" actually removes the authentication headers, so the fetch call for export is done on the same page, and the download link is created upon success. Included a success toast too! Most of the changes are on the frontend-data-set module, so I wanted to ask if this looked ok before I asked Tibor to try it out and frontend infra to review. 

Thanks!

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 1]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
 
 		window.open(url);
 	}
+	else if (target === 'download') {
+		event.preventDefault();
+
+		setLoading(true);
+		let filename = '';
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:118]: New line. These lines are unrelated.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 2]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
...
+
+		executeAsyncAction(url, method)
+			.then((response) => {
+				const disposition = response.headers.get('Content-Disposition');
+
+				if (disposition && disposition.indexOf('attachment') !== -1) {
+					const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
+					const matchesRegex = filenameRegex.exec(disposition);
+					if (matchesRegex !== null && matchesRegex[1]) {
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:127]: New line so we know that the order of these shouldn't be changed.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 3]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
...
+
+		setLoading(true);
+		let filename = '';
+
+		executeAsyncAction(url, method)
+			.then((response) => {
+				const disposition = response.headers.get('Content-Disposition');
+
+				if (disposition && disposition.indexOf('attachment') !== -1) {
+					const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:125]: Can you add docs on what these regexes do? Regex is pretty unreadable what it's trying to achieve if it needed to be modified.

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 4]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
...
+				});
+
+				setLoading(false);
+			})
+			.catch((_) => {
+				openToast({
+					message: Liferay.Language.get('unexpected-error'),
+					type: 'danger',
+				});
+				setLoading(false);
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:165]: New line

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 5]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
...
+				const downloadElement = document.createElement('a');
+
+				if (
+					filename &&
+					typeof downloadElement.download !== 'undefined'
+				) {
+					downloadElement.href = downloadUrl;
+					downloadElement.download = filename;
+					document.body.appendChild(downloadElement);
+					downloadElement.click();
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:145]: Add new lines where things shouldn't be sorted.

Alphabetize href and download.

```javascript
downloadElement.download = filename;
downloadElement.href = downloadUrl;

document.body.appendChild(downloadElement);

downloadElement.click();
```

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 6]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
...
+					if (matchesRegex !== null && matchesRegex[1]) {
+						filename = matchesRegex[1].replace(/['"]/g, '');
+					}
+				}
+
+				return response.blob();
+			})
+			.then((responseBlob) => {
+				const downloadUrl = URL.createObjectURL(responseBlob);
+				const downloadElement = document.createElement('a');
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:136]: Alphabetize

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js [hunk 7]
```diff
@@ -111,6 +111,60 @@ export function handleAction(
...
+				openToast({
+					message:
+						successMessage ||
+						Liferay.Language.get('action-completed'),
+					type: 'success',
+				});
+
+				setLoading(false);
+			})
+			.catch((_) => {
```
Comment (thektan) [inline modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:160]: If this is not being used can we remove the `_`?

**Review by thektan — CHANGES_REQUESTED** [review]
Don't forget about the formatting or else things could get improperly formatted if someone else goes in to format it. 🙂

Also since these are changes to frontend-data-set-web, are these new changes they need to review?

File: modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js (hunk 7 above)
Comment (oliv-yu) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:160]: I'm not sure, I borrowed this from the async case which did the same thing.

Comment (thektan) [reply modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/data_renderers/ActionsDropdownRenderer.js:160]: I'd remove it since it doesn't serve any purpose imo. It'd make more sense if you were accessing a second attribute like `(_, message)` and you weren't using the first value and it needed to be defined.

**Comment by oliv-yu** [comment]
Sent to frontend-infra https://github.com/liferay-frontend/liferay-portal/pull/1756
